spec/unit/util/autoload_spec.rb

```diff
@@ -198,7 +197,25 @@ def with_libdir(libdir)
     end
 
     it "should require the full path to the file" do
-      expect(Kernel).to receive(:load).with(make_absolute("/a/foo/file.rb"), any_args)
+      expect(Kernel).to receive(:load).with(path, any_args)
+
+      @autoload.loadall(env)
+    end
+
+    it "autoloads from a directory whose ancestor is Windows 8.3", if: Puppet::Util::Platform.windows? do
+      # File.expand_path will expand ~ in the last directory component only(!)
+      # so create an ancestor directory with a long path
+      dir = File.join(tmpdir('longpath'), 'short')
+      path = File.join(dir, @autoload.path, 'file.rb')
+
+      FileUtils.mkdir_p(File.dirname(path))
+      FileUtils.touch(path)
+
+      dir83 = File.join(File.dirname(basedir), 'longpa~1', 'short')
```

Comment:
should be `File.dirname(dir)` as we're not using `basedir` in this test

Reply:
ah nvm, it needs to be `basedir` so that `basedir/longpa~1` is the equivalent to `basedir/longpath`